src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_Negate.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static partial class SyntaxGeneratorExtensions
+    {
+        private const string LongLength = "LongLength";
```

Comment:
all this code was simply moved out of the invert-if feature into a shared helper.  I didn't meaningfully change it (i.e. it was all purely mechanical stuff like making static, etc.).